### DISCUSSION TITLE
Can kyuubi engine set different resources?

### DISCUSSION THREAD

**Discussion by xcl1102** [apache/kyuubi#6505] [General]
Can I set up unused yarn resources for engines of different users?
For example, using the spark engine, can I set different executor numbers and memory for user1 and user2?

**Comment by pan3793** [comment]
I'm not sure what you mean "unused yarn resources".

By default, Kyuubi uses the USER share level, which means each user creates their own Spark application.

```
bin/beeline -u jdbc:kyuubi://kyuubi:10009 -n tom \
  --hiveconf spark.executor.memory=2g
```
```
bin/beeline -u jdbc:kyuubi://kyuubi:10009 -n jerry \
  --hiveconf spark.executor.memory=8g
```

With the above commands, Tom and Jerry create their own Spark application with specific executor memory.

**Reply by pan3793** [reply]
BTW, to utilize the resource of your YARN cluster, [DRA](https://kyuubi.readthedocs.io/en/master/deployment/spark/dynamic_allocation.html) is recommended instead of using the fixed executors.

**Reply by xcl1102** [reply]
Is it possible to set different spark.dynamicAllocation.maxExecutors for each engine?  Set up more executors for users who perform large tasks, and set up a small number of executors for users who perform short jobs to reduce resource utilization.

**Reply by pan3793** [reply]
you can pass any legal Spark configurations on creating engines.

**Reply by xcl1102** [reply]
thanks you
